certora/specs/rewardsPreserve.spec

```diff
@@ -0,0 +1,186 @@
+import "methods_base.spec"
+
+
+using AToken as _AToken 
+using DummyERC20_rewardToken as _DummyERC20_rewardToken
+using DummyERC20_aTokenUnderlying as _DummyERC20_aTokenUnderlying 
+using RewardsControllerHarness as _RewardsController
+using TransferStrategyHarness as _TransferStrategyHarness
+
+
+methods
+{
+	// envfree
+	// -------
+	getUnclaimedRewards(address, address) returns (uint256) envfree
+    rewardTokens() returns (address[]) envfree
+	isRegisteredRewardToken(address) returns (bool) envfree
+    
+	// Getters from munged/harness
+    getRewardTokensLength() returns (uint256) envfree 
+    getRewardToken(uint256) returns (address) envfree
+
+	// AToken
+	_AToken.UNDERLYING_ASSET_ADDRESS() returns (address) envfree
+
+	// Reward token
+	_DummyERC20_rewardToken.balanceOf(address) returns (uint256) envfree
+	_DummyERC20_rewardToken.totalSupply() returns (uint256) envfree
+
+	// RewardsController
+    _RewardsController.getAvailableRewardsCount(address) returns (uint128) envfree
+    _RewardsController.getRewardsByAsset(address, uint128) returns (address) envfree
+	_RewardsController.getAssetListLength() returns (uint256) envfree
+	_RewardsController.getAssetByIndex(uint256) returns (address) envfree
+
+	// Permit
+	// ------
+    permit(address,address,uint256,uint256,uint8,bytes32,bytes32) => NONDET
+
+	// AToken
+	// ------
+    mint(address,address,uint256,uint256) returns (bool) => DISPATCHER(true)
+    burn(address,address,uint256,uint256) returns (bool) => DISPATCHER(true)
+}
+ 
+/**
+ * @title Single reward setup
+ * Setup the `StaticATokenLM`'s rewards so they contain a single reward token
+ * which is` _DummyERC20_rewardToken`.
+ */
+function single_RewardToken_setup() {
+	require isRegisteredRewardToken(_DummyERC20_rewardToken);
```

Comment:
is this needed now with their latest fix? i believe it's there for the claim, but now it is checked...